### PR TITLE
Feature/improve release docs

### PR DESCRIPTION
**Pull request by dawiddylowicz**
# Problem

* Release guidelines were a bit confisuing at points
* Changelog was missing release dates and link to 2.3.0 version
* Manual regression tests were missing test for testing in private mode

# Solution

* Improved release guidelines
* Updated Changelog
* Updated manual regression tests

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [n/a] Have new automated tests been implemented?
- [x] Have new manual tests been written down?
- [n/a] Have tests passed locally?
- [n/a] Have any new strings been translated?

### CODE REVIEW

File: MANUAL_REGRESSION.md [hunk 1]
```diff
@@ -32,7 +32,7 @@
     - user should be able to retake or continue with taken photo
 
 ##### 3. Cross-device with link
-(on Firefox, Safari, IE11 and Microsoft Edge browsers)
+(on private mode of Firefox, Safari, IE11 and Microsoft Edge browsers)
```
Comment (rfreitas) [inline MANUAL_REGRESSION.md]: safari was the problematic one btw. I believe mean private mode for all of the browser, but it might not be explicit.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-328-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-328-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://328-pr-onfido-sdk-ui-onfido.surge.sh

File: MANUAL_REGRESSION.md (hunk 1 above)
Comment (dawiddylowicz) [reply MANUAL_REGRESSION.md]: good point, I'll rewrite

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-328-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by dawiddylowicz** [comment]
Conflicts resolved

**Pull request merged**